<?xml version="1.0"?>
<launch>

        <param name="use_sim_time" value="true" />
    
        <node pkg="lidar_eskf" type="lidar_eskf_node" name="lidar_eskf" output="screen">
	
        <remap from="/map_update"              to="/local_map"/>
        <remap from="/lidar_eskf/cloud"        to="/dji_sim/laser/pointcloud"/>
        <remap from="/imu"                     to="/dji_sim/imu"/>
	    
        <param name="map_file_name"            value="$(find lidar_eskf)/map/tunnel.bt"/>
        <param name="octree_resolution"        value="0.1"/>
        <param name="max_obstacle_dist"        value="2.5"/>
        <param name="ray_sigma"                value="1.0"/>
        <param name="cloud_resolution"         value="0.3"/>
        <param name="cloud_range"              value="30.0"/>
        <param name="laser_type"               value="pointcloud"/>
        <param name="set_size"                 value="500"/>

        <param name="robot_frame"               value="/base_link"/>
        <param name="imu_frame"                 value="/base_link"/>
        <param name="imu_frequency"             value="100"/>
        <param name="imu_enabled"               value="true"/>
        <param name="imu_has_quat"              value="true"/>
        <param name="imu_transform"             value="true"/>
        <param name="sigma_acceleration"        value="2.0"/>
        <param name="sigma_gyroscop"            value="0.1"/>
        <param name="sigma_acceleration_bias"   value="0.002"/>
        <param name="sigma_gyroscope_bias"      value="0.001"/>
        <param name="gravity"                   value="9.8"/>
        
        <param name="init_bias_acc_x"           value="0.0"/>
        <param name="init_bias_acc_y"           value="0.0"/>
        <param name="init_bias_acc_z"           value="0.0"/>
        
        <param name="acc_queue_size"            value="3"/>
        
        <param name="imu_to_laser_roll"         value="0.0"/>
        <param name="imu_to_laser_pitch"        value="0.0"/>
        <param name="imu_to_laser_yaw"          value="0.0"/>

        <param name="smooth_enabled"            value="true"/>
        <param name="smooth_buf_size"           value="3"/>
        <param name="smooth_type"               value="median"/>
        <param name="init_roll"                 value="0.0"/>
        <param name="init_pitch"                value="0.0"/>
        <param name="init_yaw"                  value="0.0"/>
        </node>
	
        <!-- <node pkg="octomap_server" type="octomap_server_node" name="octomap_server_eskf" 
                args="$(find lidar_eskf)/map/tunnel.bt">
              <param name="frame_id"           value="world"/>
              <param name="base_frame_id"      value="world_view"/>
        </node> -->
	
        <!-- <node pkg="rviz" type="rviz" name="gpf_test_rviz" required="false" args="-d $(find lidar_eskf)/rviz/lidar_eskf.rviz"/>-->
</launch>
